tests/Integration/Behaviour/Features/Context/Domain/Product/AbstractProductFeatureContext.php

```diff
@@ -84,13 +86,17 @@ protected function assertBoolProperty(ProductForEditing $productForEditing, arra
     {
         if (isset($data[$propertyName])) {
             $expectedValue = PrimitiveUtils::castStringBooleanIntoBoolean($data[$propertyName]);
+            // Don't cast on purpose, the value should already be typed as bool
             $actualValue = $this->extractValueFromProductForEditing($productForEditing, $propertyName);
-            Assert::assertEquals(
+
+            // Use assertSame (not assertEquals) to check value AND type
+            Assert::assertSame(
                 $expectedValue,
                 $actualValue,
                 sprintf('Expected %s "%s". Got "%s".', $propertyName, $expectedValue, $actualValue)
             );
 
+            // Unset the checked field from array so we can validate they havel all been asserted
             unset($data[$propertyName]);
```

Comment:
I wonder if we could somehow implement some general assert at the end like `assertAllPropertiesHaveBeenAsserted(array $data)`.
Anyway its for some time later.

Reply:
Since it's used in many different contexts it would probably be useful yes!